Consistence/Sniffs/Exceptions/ExceptionDeclarationSniff.php

```diff
@@ -46,6 +49,43 @@ public function process(PhpCsFile $phpcsFile, $classPointer)
 				$className
 			), $classPointer, self::CODE_NOT_ENDING_WITH_EXCEPTION);
 		}
+
+		$this->checkThatExceptionIsChainable($phpcsFile, $classPointer);
+	}
+
+	/**
+	 * @param \PHP_CodeSniffer\Files\File $phpcsFile
+	 * @param int $classPointer
+	 */
+	private function checkThatExceptionIsChainable(PhpCsFile $phpcsFile, int $classPointer)
+	{
+		$functionPointerToScan = $classPointer;
+		while (($functionPointer = TokenHelper::findNext($phpcsFile, T_FUNCTION, $functionPointerToScan)) !== null) {
+			$functionName = FunctionHelper::getName($phpcsFile, $functionPointer);
+			if ($functionName === '__construct') {
+
+				$typeHints = FunctionHelper::getParametersTypeHints($phpcsFile, $functionPointer);
+				if (count($typeHints) === 0) {
+					$phpcsFile->addError(
+						'Exception is not chainable. It must have optional \Throwable as last constructor argument.',
+						$functionPointer,
+						self::CODE_NOT_CHAINABLE
+					);
+					return;
+				}
+				$lastArgument = array_pop($typeHints);
+				if ($lastArgument->getTypeHint() !== '\Throwable') {
+					$phpcsFile->addError(sprintf(
+						'Exception is not chainable. It must have optional \Throwable as last constructor argument and has "%s".',
```

Comment:
Can you please add a test for the case when the last parameter has no type hint?

Reply:
nice catch!